ntoskrnl/mm/rmap.c

```diff
@@ -158,10 +159,15 @@ MmPageOutPhysicalAddress(PFN_NUMBER Page)
 
         /* Delete this virtual mapping in the process */
         MmDeleteRmap(Page, Process, Address);
-        MmDeleteVirtualMapping(Process, Address, &Dirty, &MapPage);
-
-        /* We checked this earlier */
-        ASSERT(MapPage == Page);
+        Unmapped = MmDeleteVirtualMapping(Process, Address, &Dirty, &MapPage);
+        if (!Unmapped || (MapPage != Page))
+        {
+            /* 
+             * Something's corrupted, we got there because this process is 
+             * supposed to map this page there.
```

Comment:
to map? or "to have mapped" ?

Reply:
to be mapping. We're in the process of unmapping it.